File: backend/src/main/java/com/ttn/sporttn/modules/product/entity/Brand.java
package com.ttn.sporttn.modules.product.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
@Table(name = "brands", indexes = {
        @Index(name = "idx_brands_slug",   columnList = "slug"),
        @Index(name = "idx_brands_active", columnList = "active")
})
public class Brand {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false, unique = true, length = 100)
    private String name;

    private String slug;

    private String description;

    private String color;

    @Column(name = "logo_url")
    private String logoUrl;

    @Column(name = "website_url")
    private String websiteUrl;

    private Boolean active = true;

    @OneToMany(mappedBy = "brand")
    private List<Product> products;
}

File: frontend/src/app/core/models/brand/brand.ts
export interface BrandAddRequest {
  name: string;
  slug: string;
  description: string;
  logoUrl: string;
  websiteUrl: string;
  active: boolean;
  color: string;
}

export interface BrandResponse {
  id: number;
  name: string;
  slug: string;
  description: string;
  logoUrl: string;
  websiteUrl: string;
  active: boolean;
  totalProducts: number;
  color: string;
}
